### PR TITLE
[SPARK-39392][SQL] Refine ANSI error messages for try_* function hints

### PR DESCRIPTION
**Pull request by vitaliili-db**
### What changes were proposed in this pull request?

Refine ANSI error messages and remove 'To return NULL instead'

### Why are the changes needed?

Improve error messaging for ANSI mode since the user may not even aware that query was returning NULLs.

### Does this PR introduce _any_ user-facing change?

No

### How was this patch tested?

Unit tests

### CODE REVIEW

**Comment by vli-databricks** [comment]
@gengliangwang please take a look

File: core/src/main/resources/error/error-classes.json [hunk 1]
```diff
@@ -195,7 +195,7 @@
   },
   "INVALID_ARRAY_INDEX_IN_ELEMENT_AT" : {
     "message" : [
-      "The index <indexValue> is out of bounds. The array has <arraySize> elements. To return NULL instead, use `try_element_at`. If necessary set <config> to \"false\" to bypass this error."
+      "The index <indexValue> is out of bounds. The array has <arraySize> elements. Use `try_element_at` to tolerate accessing element at non-existing index and return NULL instead. If necessary set <config> to \"false\" to bypass this error."
```
Comment (gengliangwang) [inline core/src/main/resources/error/error-classes.json:198]: `non-existing index` => `invalid index` ?

**Review by gengliangwang — APPROVED** [review]

**Comment by gengliangwang** [comment]
cc @srielau as well

File: core/src/main/resources/error/error-classes.json (hunk 1 above)
Comment (vli-databricks) [reply core/src/main/resources/error/error-classes.json]: Yes, that is better. Fixed.

**Review by HyukjinKwon — APPROVED** [review]

**Review by MaxGekk — APPROVED** [review]

**Comment by MaxGekk** [comment]
+1, LGTM. Merging to master.
Thank you, @vli-databricks and @gengliangwang @HyukjinKwon for review.

**Comment by MaxGekk** [comment]
@vli-databricks Could you backport the changes to branch-3.3, please.

**Pull request closed without merging**